pkg/kv/kvserver/stateloader/initial_test.go

```diff
@@ -7,19 +7,57 @@ package stateloader
 
 import (
 	"context"
+	"path/filepath"
 	"reflect"
+	"strings"
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/print"
 	"github.com/cockroachdb/cockroach/pkg/raft/raftpb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/echotest"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/stretchr/testify/require"
 )
 
+// TestWriteInitialRangeState captures the typical initial range state written
+// to storage at cluster bootstrap.
+func TestWriteInitialRangeState(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	eng := storage.NewDefaultInMemForTesting()
+	defer eng.Close()
+	b := eng.NewBatch() // TODO(pav-kv): make it write-only batch
+	defer b.Close()
+
+	desc := roachpb.RangeDescriptor{
+		RangeID:       5,
+		StartKey:      roachpb.RKey("a"),
+		EndKey:        roachpb.RKey("z"),
+		NextReplicaID: 2,
+	}
+	const replicaID = roachpb.ReplicaID(1)
+	// Use PreviousRelease so that we don't have to update this test on every
+	// version gate addition.
+	cv := clusterversion.ClusterVersion{Version: clusterversion.PreviousRelease.Version()}
```

Comment:
TODO: find a version handle that doesn't change or get removed. Or maybe just use zero.